resources/sugarcrm/emailshandler.cpp

```diff
@@ -132,12 +132,14 @@ void EmailsHandler::getExtraInformation(Akonadi::Item::List &items)
             if (pos == -1) {
                 qCWarning(FATCRM_SUGARCRMRESOURCE_LOG) << "Email not found:" << email_id;
             } else {
-                SugarEmail email = items[pos].payload<SugarEmail>();
+                auto item = items.at(pos);
```

Comment:
I'm never sure with auto: does this make a copy or does that declare item as a ref, given that at() returns a ref?

If it makes a copy, then this patch is wrong, due to item.setPayload() below only affecting the copy.

Looks good otherwise (I hate exceptions because it's harder to track where they come from when they're uncaught... I thought we had such asserts everywhere, but maybe that was only in the fatcrm client then, not in the resource. These asserts are therefore very welcome).